desc/optimize/optimizer.py

```diff
@@ -141,6 +145,13 @@ def optimize(  # noqa: C901 - FIXME: simplify this
             `OptimizeResult` for a description of other attributes.
 
         """
+        things = flatten_list(things, flatten_tuple=True)
+
+        # need local import to avoid circular dependencies
+        from desc.equilibrium import Equilibrium
+
+        eq = get_instance(things, Equilibrium)
```

Comment:
this line makes it seem we ALWAYS require an equilibrium when we want to optimize things. What if we just want to minimize quadratic flux from coils on a surface (not necessarily an equilibirum surface)?

Reply:
It looks like this logic is to handle the case of proximal projections, and I think your example of not needing an equilibrium would still work fine. But I agree it seems misleading.

Reply:
I'll add a comment to clarify